src/pm-conversations/pmConversationsSelectors.js

```diff
@@ -1,46 +1,62 @@
 /* @flow strict-local */
 import { createSelector } from 'reselect';
 
-import type { Message, PmConversationData, Selector, User } from '../types';
+import type {
+  Message,
+  PmConversationData,
+  RecentPrivateConversation,
+  Selector,
+  User,
+  UserOrBot,
+} from '../types';
+import { getServerVersion } from '../account/accountsSelectors';
 import { getPrivateMessages } from '../message/messageSelectors';
-import { getOwnUser } from '../users/userSelectors';
+import { getRecentPrivateConversations } from '../directSelectors';
+import { getOwnUser, getAllUsersById } from '../users/userSelectors';
 import { getUnreadByPms, getUnreadByHuddles } from '../unread/unreadSelectors';
 import { normalizeRecipientsSansMe, pmUnreadsKeyFromMessage } from '../utils/recipient';
+import { ZulipVersion } from '../utils/zulipVersion';
 
-export const getRecentConversations: Selector<PmConversationData[]> = createSelector(
-  getOwnUser,
-  getPrivateMessages,
-  getUnreadByPms,
-  getUnreadByHuddles,
-  (
-    ownUser: User,
-    messages: Message[],
-    unreadPms: { [number]: number },
-    unreadHuddles: { [string]: number },
-  ): PmConversationData[] => {
-    const recipients = messages.map(msg => ({
-      ids: pmUnreadsKeyFromMessage(msg, ownUser.user_id),
-      emails: normalizeRecipientsSansMe(msg.display_recipient, ownUser.email),
-      msgId: msg.id,
-    }));
+/**
+ * Given a list of PmConversationPartial or PmConversationData, trim it down to
+ * contain only the most recent message from any conversation, and return them
+ * sorted by recency.
+ */
+const collateByRecipient = <T: { recipients: string, msgId: number }>(
```

Comment:
Looks like this is only used in the one place? I don't think it's a win to pull it out -- the interface is pretty complicated in isolation.